lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,28 @@
+module Semian
+  class AtomicInteger #:nodoc:
+    attr_accessor :value
+
+    def initialize(_name, _permissions)
+      @value = 0
+    end
+
+    def increment_by(val)
+      self.value += val
+    end
+
+    def increment
+      increment_by(1)
+    end
+
+    def execute_atomically
+      yield if block_given?
+    end
+
+    def shared?
+      false
+    end
```

Comment:
The user should care because they might want to handle the two cases that can occur (shared, not shared) separately. 

Easier to test and less repeated code through something like this:
```ruby
def test_shared_attribute
  run_test_with_sliding_window_classes do |klass, instance|
    return unless instance.shared?
    # test contents
  end
end
```
There is no point in writing the same code x times, once for each type of class and once for a shared vs not-shared instance of a class. 

How else can you distinguish between a shared-memory version of a class and a not-shared-memory version? There is no other method that can distinguish this without explicitly testing to see if it is. When the `SharedMemoryObject` superclass comes in, `shared?`, `execute_atomically`, and `destroy` will be moved there. 

I'm pretty adamant about this being included since many later tests for testing the shared component depend on this as well and otherwise there is no way to distinguish them making writing tests more difficult than it should be. If the issue is that it doesn't make sense to include it at this iteration, but that it will be included when I add in the shared memory class, then that's OK.